core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java

```diff
@@ -522,9 +522,6 @@ public void resetWorkflow(String workflowId) {
     public List<Task> createTasks(List<Task> tasks) {
         List<Task> tasksCreated = new LinkedList<>();
         tasksCreated = executionDAO.createTasks(tasks);
-        for (Task task : tasksCreated) {
```

Comment:
It will be better for this feature to be enabled by a flag. Flat out removing this code - will make future merges hard. We will not know if this is new code added since our last merge. 
Can we introduce a new system property - indexOnFailure which we set to enable. if it is not set- it defaults to older behavior.

Reply:
added a flag named indexOnCommunicationFailure